drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -823,11 +879,52 @@ static int udc_mcux_driver_preinit(const struct device *dev)
 	data->caps.hs = true;
 	priv->dev = dev;
 
+
 	pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
 
 	return 0;
 }
 
+static int udc_mcux_driver_preinit(const struct device *dev)
+{
+	const struct udc_mcux_config *config = dev->config;
+	struct udc_data *data = dev->data;
+	struct udc_mcux_data *priv = data->priv;
+
+	udc_mcux_get_hal_driver_id(priv, config);
+	if (priv->controller_id == 0xFFu) {
+		return -ENOMEM;
+	}
+
+	k_mutex_init(&data->mutex);
+	k_fifo_init(&priv->fifo);
+	k_work_init(&priv->work, udc_mcux_work_handler);
+	return udc_mcux_init_common(dev);
+}
+
+#ifdef CONFIG_PM
+static int udc_mcux_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	struct udc_mcux_data *priv = udc_get_private(dev);
+
```

Comment:
Apart from whether PM usage makes sense or not, it should be like
```c
	if (IS_ENABLED(CONFIG_PM)) {
		return -ENOTSUP;
	}
```
if not possible, then it should be fixed in PM subsytem.

Reply:
@jfischer-no This function needs to be wrapped with CONFIG_PM_DEVICE because of the way the PM macros work. At the bottom, PM_DEVICE_DT_INST_DEFINE() is setting things up IF the application has been defined for PM_DEVICE. If it hasn't, then the PM_DEVICE_DT_INST_DEFINE() is a null and to avoid "unused function" we need to wrap this overall function with the #if.

Reply:
This is what pm_device_driver_init is for, it allows you to reuse the pm hook implementations if PM_DEVICE is n

Reply:
@jfischer-no No what? Use your words. You did not respond to the point that @dleach02 actually made. Are you claiming that what he said is not true?

Reply:
Statement `"If it hasn't, then the PM_DEVICE_DT_INST_DEFINE() is a null and to avoid "unused function" we need to wrap this overall function with the #if."` is false.
Therefore my answer is 'No.', followed by a proof inside a diff that can be applied to the commit.